payment_methods/Paypal_Standard/ItemizedOrder.php

```diff
@@ -0,0 +1,327 @@
+                \EEH_Debug_Tools::printr($this->itemized_order_sum, '$this->itemized_order_sum', __FILE__, __LINE__);
+                $item_num++;
+            }
+        }
+        return $item_num;
+    }
+
+
+    /**
+     * @param EE_Line_Item $total_line_item
+     * @param int          $item_num
+     * @return int
+     * @throws EE_Error
+     * @throws ReflectionException
+     */
+    private function handleItemizedOrderSumDifference(EE_Line_Item $total_line_item, int $item_num): int
+    {
+        \EEH_Debug_Tools::printr(__FUNCTION__, __CLASS__, __FILE__, __LINE__, 2);
+        $taxes_li = EEH_Line_Item::get_taxes_subtotal($total_line_item);
+        // calculate the difference between the TXN total and the itemized order sum
+        $itemized_order_sum_difference = round(
+            $this->transaction->total()
+            - $this->itemized_order_sum
+            - $taxes_li->total()
+            - $this->existing_shipping_charges,
+            2
+        );
+        \EEH_Debug_Tools::printr($this->transaction->total(), '$this->transaction->total()', __FILE__, __LINE__);
+        \EEH_Debug_Tools::printr($this->transaction->paid(), '$this->transaction->paid()', __FILE__, __LINE__);
+        \EEH_Debug_Tools::printr($this->itemized_order_sum, '$this->itemized_order_sum', __FILE__, __LINE__);
+        \EEH_Debug_Tools::printr($taxes_li->total(), '$taxes_li->total()', __FILE__, __LINE__);
+        \EEH_Debug_Tools::printr(
+            $this->existing_shipping_charges,
+            '$this->existing_shipping_charges',
+            __FILE__,
+            __LINE__
+        );
+        \EEH_Debug_Tools::printr(
+            $itemized_order_sum_difference,
+            '$itemized_order_sum_difference',
+            __FILE__,
+            __LINE__
+        );
+        // ideally the itemized order sum equals the transaction total, but if not (which is weird),
+        // and the itemized sum is MORE than the transaction total...
+        if (EEH_Money::compare_floats($itemized_order_sum_difference, 0.00, '<')) {
+            echo "\n\n~~~~~~~~~~~~~~~ ADD DISCOUNT ?!?!?! ~~~~~~~~~~~~~~~\n\n";
+
+            // add the difference to the discounts
+            $this->total_discounts += abs($itemized_order_sum_difference);
+        } elseif (EEH_Money::compare_floats($itemized_order_sum_difference, 0.00, '>')) {
+            // the itemized order sum is LESS than the transaction total
+            $this->order_items["item_name_{$item_num}"] = substr(
+                esc_html__('additional charges', 'event_espresso'),
+                0,
+                127
+            );
+            $this->order_items["amount_{$item_num}"]    = $this->gateway_data_formatter->formatCurrency(
+                $itemized_order_sum_difference
+            );
+            $this->order_items["quantity_{$item_num}"]  = 1;
+            $item_num++;
+        }
+        if (EEH_Money::compare_floats($this->total_discounts, 0.00, '>')) {
+
```

Comment:
(huh, no more code suggestions ?)

Unnecessary empty line.

Reply:
> (huh, no more code suggestions ?)

that's because this was already fixed but GitHub was still showing the previous version.

it _should_ show you something like this though to indicate that a refresh is needed:
<img width="737" alt="Screenshot 2021-12-03 094829" src="https://user-images.githubusercontent.com/1751030/144649200-115073bd-4345-417a-98a2-fbc22c772f46.png">